### PR TITLE
Fix installcheck, rescue mode & co. on Xen & Hyper-V and align svirt booting with QEMU

### PR DESCRIPTION
**Pull request by mnowaksuse**
Validation runs:
* Tumbleweed QEMU `mediacheck`: http://assam.suse.cz/tests/855
* SLES Xen PV `installcheck`: http://assam.suse.cz/tests/858
* SLES Xen HVM `installcheck`: http://assam.suse.cz/tests/857
* SLES Xen HVM `ext4`: http://assam.suse.cz/tests/856
* SLES Hyper-V `mediacheck`: http://assam.suse.cz/tests/859
* SLES Xen HVM `memtest`: http://assam.suse.cz/tests/860

### CODE REVIEW

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -197,6 +197,8 @@ sub select_bootmenu_more {
     else {
         send_key_until_needlematch($tag, 'down', 10, 5);
     }
+    # Hyper-V defaults to 1280x1024, we need to fix it here
+    type_string 'video=hyperv_fb:1024x768' if check_var('VIRSH_VMM_FAMILY', 'hyperv');
```
Comment (okurz) [inline lib/opensusebasetest.pm]: Hm, commonly we set the video resolution boot parameters within lib/bootloader_setup.pm . Could you use a helper function from there?

File: products/sle/main.pm [hunk 2]
```diff
@@ -1182,6 +1182,12 @@ elsif (get_var("RESCUESYSTEM")) {
     loadtest "installation/rescuesystem_validate_sle";
 }
 elsif (get_var("INSTALLCHECK")) {
+    if (check_var('VIRSH_VMM_FAMILY', 'hyperv')) {
+        loadtest "installation/bootloader_hyperv";
+    }
+    else {
+        loadtest "installation/bootloader_svirt";
+    }
```
Comment (okurz) [inline products/sle/main.pm]: Hm, when other backends already work without a specific bootloader test module before "rescuesystem" how about implementing bootloader helper functions or using existing ones from lib/bootloader_setup.pm and using them where necessary within installation/rescuesystem as in tests/boot/boot_do_desktop.pm ?

File: lib/opensusebasetest.pm (hunk 1 above)
Comment (mnowaksuse) [reply lib/opensusebasetest.pm]: Now there's a more generic approach to setting GFX.

File: products/sle/main.pm (hunk 2 above)
Comment (mnowaksuse) [reply products/sle/main.pm]: I untangled VM setup and mangling with GRUB/sys-inst, so running "unusual" scenarios on svirt needs to call just one sub.

**Comment by okurz** [comment]
much nicer. Thank you, appreciated.

**Comment by coolo** [comment]
https://openqa.suse.de/tests/1249950 regressed because of this - please fix

**Comment by mnowaksuse** [comment]
I am sorry for the regression, here's the fix: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3893.

**Pull request merged**